### ISSUE TITLE
Build error after upgrading from kustomize 4.5.4 to 4.5.5

### ISSUE THREAD

**Issue by bwinter** [kubernetes-sigs/kustomize#4678]
**Describe the bug**

I am running the following command:
```
docker run -v "$(pwd):/tmp" k8s.gcr.io/kustomize/kustomize:v4.5.5 build "https://github.com/kubernetes-sigs/aws-efs-csi-driver.git//deploy/kubernetes/overlays/stable?ref=release-1.3&timeout=90s"
```
It fails almost immediately with:
```
Error: git cmd = '/usr/bin/git remote add origin https://github.com/kubernetes-sigs/aws-efs-csi-driver.git': exit status 128
```

I think the important thing to note is this works:
```
docker run -v "$(pwd):/tmp" k8s.gcr.io/kustomize/kustomize:v4.5.4 build "https://github.com/kubernetes-sigs/aws-efs-csi-driver.git//deploy/kubernetes/overlays/stable?ref=release-1.3&timeout=90s"
```

Same command but kustomize patch version changed.

**Expected output**

I expect it to build the full yaml output for the csi driver.

**Actual output**

> Error: git cmd = '/usr/bin/git remote add origin https://github.com/kubernetes-sigs/aws-efs-csi-driver.git': exit status 128

**Kustomize version**

Can see these in the docker run command parameters.

**Platform**

macOS / Docker

**Comment by KnVerey** [comment]
/triage not-reproducible
/close

I was able to run the exact 4.5.5 command provided without error. The error quoted sounds like it is coming from git itself, so perhaps it was a transient issue with Github, or the container's connectivity to it. Please reopen if you are able to reproduce consistently.

**Comment by bwinter** [comment]
@KnVerey - that is unusual.

I am back to try this update again and am still having the same problem.

Running the following with these results:
```
$ docker run -v "$(pwd):/tmp" "k8s.gcr.io/kustomize/kustomize:v4.5.5"   build "https://github.com/kubernetes-sigs/aws-ebs-csi-driver.git//deploy/kubernetes/overlays/stable/?ref=release-1.11&timeout=90s"   -o "/tmp/ebs-csi-driver.yaml"
Error: git cmd = '/usr/bin/git remote add origin https://github.com/kubernetes-sigs/aws-ebs-csi-driver.git': exit status 128
```

Where as, again with 4.5.4 it seems to work.

And I can run it as many times as I want and it seems to continue to fail each run.

I am running this on a MacBook Pro. Montery 12.5

git version is 2.36.1

Using iterm / bash

Have a variety of terminal tools installed.

I am using colima.

Hmm, what else might be relevant? 🤔

**Comment by KnVerey** [comment]
> Also not sure how to reopen.

`/reopen` will do it, but unfortunately I still can't reproduce

I tried several more times with _exactly_ the same command you're using, and it always works for me. I also tried shelling into the container (`docker run -i -t  --entrypoint /bin/sh k8s.gcr.io/kustomize/kustomize:v4.5.5`) and running relevant commands directly in there (e.g. `git clone --branch=release-1.11 https://github.com/kubernetes-sigs/aws-ebs-csi-driver` then `git remote remove origin && git remote add origin https://github.com/kubernetes-sigs/aws-ebs-cs
i-driver.git` in the subdir, then `./kustomize build aws-ebs-csi-driver/deploy/kubernetes/overlays/stable` in the app dir) and everything worked fine. Perhaps try something like that? Maybe you'll get a more informative error message.

Have you tried newer versions of the container?
